modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.tsx

```diff
@@ -133,12 +138,48 @@ const FrontendDataSetContent = ({
 	showSelectAll = false,
 	sidePanelId,
 	sorts: sortsProp = [],
+	stateInURLSettings = EStateInURLSettings.OFF,
 	style = 'default',
 	uniformActionsDisplay,
 	views,
 }: IFrontendDataSetProps) => {
 	const fdsRef = useRef(null);
 	const dataSetWrapperRef: RefObject<HTMLDivElement> = useRef(null);
+
+	const [getDelta, setDelta] = useStateInURL({
+		id,
+		stateDispatcher: {
+			key: EStateInURLKeys.DELTA,
+			type: VIEWS_ACTION_TYPES.UPDATE_PAGINATION_DELTA,
+		},
+		stateInURLSettings,
+		stateInitializer: (delta: number) => {
+			if (isNaN(delta) || delta < 1) {
+				return undefined;
+			}
+
+			return delta;
+		},
+	});
```

Comment:
Much nicer! 

Does `getDelta` need to be a function? Can it be just a variable `delta`? In the file, I only see it used with immediate invocation, `getDelta()`. If we are not using it somewhere strictly as a function, and if it doesn't have any arguments, I believe it can just be a variable.

Reply:
It needs to, because each time we invoke it, its value is taken from the URL, so it might have changed across calls. Theoretically, one does not need to rerender the entire FDS upon delta change so hook does not run again.

Reply:
Still, we can optimize the call to avoid reading the URL so many times...